kyaml/fn/framework/result.go

```diff
@@ -103,7 +103,13 @@ type Field struct {
 	// CurrentValue is the current field value
 	CurrentValue interface{} `yaml:"currentValue,omitempty" json:"currentValue,omitempty"`
 
+	// ProposedValues is a set of proposed value of the field to fix an issue.
+	ProposedValues []interface{} `yaml:"proposedValues,omitempty" json:"proposedValues,omitempty"`
+
 	// ProposedValue is the proposed value of the field to fix an issue.
+	//
+	// Note: This field will be removed on KRM Spec config.kubernetes.io/v2
+	// The value gets mapped into ProposedValues.
 	ProposedValue interface{} `yaml:"proposedValue,omitempty" json:"proposedValue,omitempty"`
```

Comment:
What do we think about defining the all the ResourceList versions separately? 

E.g. have separate files `kyaml/fn/framework/types/v1/resourcelist.go` and `kyaml/fn/framework/types/v2alpha1/resourcelist.go`, where each has its own definitions? I feel like having all the possible fields across all the versions in each struct will be difficult to keep track of, especially if we ever have multiple changing fields.

Reply:
cc @KnVerey

Reply:
I also thought this would be ideal, that would require however many changes across the libraries, no?
e.g. because these are not interfaces and because Go static typing it'd require a lot of code duplication (unless we choose to use go1.18 generics 😛 )

but maybe I don't understand the full scope and maybe it's not as large as I'm imagining 🤔

Reply:
Ah yeah I think you're right, it would involve a lot of code changes across the library and potentially code duplication. I suppose this will have to do unless we want to do a major overhaul of kyaml.

In the comments, could you explicitly label both `ProposedValue` and `ProposedValues` with what versions of ResourceList each is supported in?